File: fixtures/scale/ruby/large/lib/check_mirrorpack.rb
require_relative 'mirrorpack_encode'
require_relative 'mirrorpack_decode'
require_relative 'corpus_codec'

def check_mirrorpack
  corpus_codec.each do |text|
    packed = mirrorpack_encode(text)
    return false if mirrorpack_decode(packed) != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_mirrorpack.py
from cipherlab.ciphers.mirrorpack import mirrorpack_encode, mirrorpack_decode
from cipherlab.specs.mirrorpack_spec import mirrorpack_spec
from cipherlab.corpus.corpus_codec import corpus_codec


def check_mirrorpack():
    spec = mirrorpack_spec()
    for text in corpus_codec():
        packed = mirrorpack_encode(text)
        if mirrorpack_decode(packed) != text:
            return False
    return spec["category"] == "codec"
